WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductItemViewHolder.kt

```diff
@@ -134,4 +148,13 @@ class ProductItemViewHolder(parent: ViewGroup) : RecyclerView.ViewHolder(
 
         return if (statusHtml != null) "$statusHtml $bullet $stock" else stock
     }
+
+    /**
+     * Method to return details associated with a user selection
+     */
+    fun getItemDetails(): ItemDetailsLookup.ItemDetails<Long> =
+        object : ItemDetailsLookup.ItemDetails<Long>() {
+            override fun getPosition(): Int = adapterPosition
+            override fun getSelectionKey(): Long? = itemId
+        }
```

Comment:
I think we can hide the repeated types declaration here to reduce the verbosity and use a little bit of the type inference, WDYT?

```suggestion
    fun getItemDetails() =
        object : ItemDetailsLookup.ItemDetails<Long>() {
            override fun getPosition() = adapterPosition
            override fun getSelectionKey() = itemId
        }
```